sources/api/apiserver/src/server/ephemeral_storage.rs

```diff
@@ -114,10 +114,17 @@ pub fn bind(variant: &str, dirs: Vec<String>) -> Result<()> {
 
     let mount_point = format!("/mnt/{}", EPHEMERAL_MNT);
     let mount_point = Path::new(&mount_point);
-    let allowed_dirs = allowed_bind_dirs(variant);
+    let (allowed_exact, allowed_prefixes, disallowed_contains) = allowed_bind_dirs(variant);
     for dir in &dirs {
+        let exact_match = allowed_exact.contains(dir.as_str());
+        let prefix_match = allowed_prefixes
+            .iter()
+            .any(|prefix| dir.starts_with(prefix));
+        let disallowed_match = disallowed_contains
+            .iter()
+            .any(|contains| dir.contains(contains));
         ensure!(
-            allowed_dirs.contains(dir.as_str()),
+            (exact_match || prefix_match) && !disallowed_match,
```

Comment:
nit: might be slightly more efficient if we assume exact matches will never be disallowed:
```suggestion
            (exact_match || (prefix_match && !disallowed_match),
```

Reply:
Thanks, done this in most latest commit.